packages/react-dialog/Spec.md

```diff
@@ -2,62 +2,303 @@
+
+The visibility of the dialog is controlled through the `hidden` prop whose its value should be a react state boolean provided from the consumer.
+
+[Documentation for v8 Dialog](https://developer.microsoft.com/en-us/fluentui#/controls/web/dialog)
+
+Sample code:
+
+```jsx
+  <DefaultButton secondaryText="Opens the Sample Dialog" onClick={toggleHideDialog} text="Open Dialog" />
+
+  <Dialog
+    hidden={hideDialog}
+    onDismiss={toggleHideDialog}
+    dialogContentProps={dialogContentProps}
+    modalProps={modalProps}
+  >
+    <DialogFooter>
+      <PrimaryButton onClick={toggleHideDialog} text="Send" />
+      <DefaultButton onClick={toggleHideDialog} text="Don't send" />
+    </DialogFooter>
+  </Dialog>
+
+```
+
+#### v0
+
+In v0, the Dialog component expects all the content through props, including the content, actions etc. The dialog component uses the `trigger` prop that expects a React component to control its visibility. The element passed to this prop will be rendered in-place where the dialog is defined.
+
+[Documentation for v0 Dialog](https://fluentsite.z22.web.core.windows.net/components/dialog/definition)
+
+```jsx
+<Dialog
+  cancelButton="Connect protocol"
+  confirmButton="Transmit capacitor"
+  content="Connect driver"
+  header="Transmit capacitor"
+  headerAction="Generate protocol"
+  trigger={<Button content="A trigger" />}
+/>
+```
+
+## API proposal
+
+Dialog component is composed by 3 sub-components: Header, Body and Footer.
+
+### Components
+
+| Component    | Purpose                                                  |
+| ------------ | -------------------------------------------------------- |
+| Dialog       | The main wrapper component.                              |
+| DialogHeader | (optional) Component for the title and the close button. |
+| DialogBody   | (optional) Component for the main content of dialog.     |
+| DialogFooter | (optional) Component for the main actions of dialog.     |
+
+## Dialog
+
+The dialog is a container which handles styling (border, background etc.) and the overlay (dimmed background). By default the rendered DOM is a portal to a div in body.
+
+### Anatomy
+
+![Visual anatomy of Dialog component](./assets/dialog-anatomy.png)
+
+### API
+
+| Property | Values                        | Default   | Purpose                                  |
+| -------- | ----------------------------- | --------- | ---------------------------------------- |
+| type     | `modal`, `non-modal`, `alert` | `modal`   | Dialog variations                        |
+| open     | boolean                       | `false`   | Set to `true` when the dialog is visible |
+| overlay  | _slot_                        | undefined | Dimmed background of dialog              |
+
+> ⚠️ _Pending issue: `draggable`, the dragging functionality of the dialog is under consideration. This might be excluded from the dialog implementation._
```

Comment:
feels like this would be a good place to demo how you can compose drag and drop functionality around the modal

Reply:
I had in mind to address the drag and drop functionality later, and/or include it in the documentation. WDYT?

Reply:
I'd say be specific in the spec that draggable is out of scope of the component, and make a note in "dialog examples" section to add a draggable example that is composed with some yet to be determined DnD library